### DISCUSSION TITLE
How to limit dragging area with a rectangle?

### DISCUSSION THREAD

**Discussion by AlexIII** [konvajs/konva#1270] [Q&A] [answered]
Would be glad to hear any ideas on how this could be implemented.

**Comment by lavrton** [comment, marked as answer]
What did you try?

**Reply by AlexIII** [reply]
Listening on the pointer move event, computing the pointer position delta, checking if it's the new position is inside the allowed area and, if it is, adjusting x and y of the Stage manually. Basically, it's re-implementation of `draggable` feature in the user-space and I don't like this at all.

**Reply by lavrton** [reply]
Why do you need to reimplement it? There are two demos with limits
https://konvajs.org/docs/drag_and_drop/Simple_Drag_Bounds.html
https://konvajs.org/docs/drag_and_drop/Complex_Drag_and_Drop.html

Can't you do that?

**Reply by AlexIII** [reply]
I can, I completely missed that. Thank you.
